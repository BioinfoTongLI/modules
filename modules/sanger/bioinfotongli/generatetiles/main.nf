process BIOINFOTONGLI_GENERATETILES {
    tag "${meta.id}"

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        "quay.io/bioinfotongli/large_image_io:0.0.2":
        "quay.io/bioinfotongli/large_image_io:0.0.2"}"

    input:
    tuple val(meta), path(image)

    output:
    tuple val(meta), path("${output_name}"), emit: tile_coords
    path "versions.yml"           , emit: versions

    script:
    stem = meta.id
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    output_name = "${prefix}_tile_coords.csv"
    """
    tile_2D_image.py run \\
        --image ${image} \\
        --output_name "${output_name}" \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bioinfotongli: \$(tile_2D_image.py version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    output_name = "${prefix}_tile_coords.csv"
    """
    touch "${output_name}"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bioinfotongli: \$(tile_2D_image.py version)
    END_VERSIONS
    """
}