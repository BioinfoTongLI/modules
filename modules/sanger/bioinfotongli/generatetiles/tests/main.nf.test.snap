{
    "mouse_heart - tiff - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_tile_coords.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,31c29985a7b71d21b5e6eb2c4eabb930"
                ],
                "tile_coords": [
                    [
                        {
                            "id": "test"
                        },
                        "test_tile_coords.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,31c29985a7b71d21b5e6eb2c4eabb930"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-02-07T13:38:28.365488933"
    },
    "mouse_heart - tiff": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_tile_coords.csv:md5,736ea44566b4ff52f7bcf1d46078152b"
                    ]
                ],
                "1": [
                    "versions.yml:md5,31c29985a7b71d21b5e6eb2c4eabb930"
                ],
                "tile_coords": [
                    [
                        {
                            "id": "test"
                        },
                        "test_tile_coords.csv:md5,736ea44566b4ff52f7bcf1d46078152b"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,31c29985a7b71d21b5e6eb2c4eabb930"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-02-07T13:38:20.806020835"
    }
}